src/Tests/RegistrationTest.php

```diff
@@ -25,19 +25,133 @@ class RegistrationTest extends WebTestBase {
   public static $modules = ['decoupled_auth', 'user', 'system'];
 
   /**
+   * The user settings config.
+   *
+   * @var \Drupal\Core\Config\Config
+   */
+  protected $user_config;
+
+  /**
+   * The acquisition settings config.
+   *
+   * @var \Drupal\Core\Config\Config
+   */
+  protected $acquisition_config;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->user_config = $this->config('user.settings');
+    $this->acquisition_config = $this->config('decoupled_auth.settings');
+  }
+
+  /**
+   * Don't require email verification and allow registration by site visitors
+   * without administrator approval.
+   */
+  protected function allowUserRegistration() {
+    $this->user_config
+      ->set('verify_mail', FALSE)
+      ->set('register', USER_REGISTER_VISITORS)
+      ->save();
+  }
+
+  /**
+   * Disable acquisition on registration.
+   */
+  protected function disableRegistrationAcquisition() {
+    $this->acquisition_config->set('acquisitions.registration', 0)->save();
+  }
+
+  /**
+   * Post user information to user register form.
+   *
+   * @param string $name
+   * @param string $mail_prefix
+   * @param string $pass
+   *
+   * @return array
+   *   Array of information that was used to create user.
+   */
+  protected function registerNewUser($name = '', $mail_prefix = '', $pass = '') {
+    $edit = array();
+    $edit['name'] = $name ? $name : $this->randomMachineName();
+    $edit['mail'] = $mail_prefix ? $mail_prefix . '@example.com' : $edit['name'] . '@example.com';
+
+    if (!is_null($pass)) {
+      $edit['pass[pass1]'] = $pass ? $pass : $this->randomMachineName();
+      $edit['pass[pass2]'] = $edit['pass[pass1]'];
+    }
+
+    $this->drupalPostForm('user/register', $edit, t('Create new account'));
+
+    return $edit;
+  }
+
+  /**
+   * Load User objects by an entity property.
+   *
+   * @param array $property
+   *   An array of a property value keyed by the property.
+   * @return array
+   *   An array of User entity objects indexed by their ids.
+   */
+  protected function getUsersByProperty(array $property) {
```

Comment:
Every call to this passes in `['name' => $edit['name']]`, so why not make it `getUserByName($name)`?